File: packages/types/src/schemas/template.ts
import { z } from 'zod';

// Template schema
export const TemplateSchema = z.object({
  name: z.string(),
  description: z.string().optional(),
  requiredHeadings: z.array(z.string()),
  defaultSections: z.array(
    z.object({
      heading: z.string(),
      placeholder: z.string(),
    })
  ),
  customFieldMappings: z.record(z.string(), z.string()).optional(),
});

export type Template = z.infer<typeof TemplateSchema>;

File: packages/gamole_types/schemas/template.py
"""Template schemas - ported from packages/types/src/schemas/template.ts."""

from pydantic import BaseModel


class TemplateSection(BaseModel):
    heading: str
    placeholder: str


class Template(BaseModel):
    name: str
    description: str | None = None
    required_headings: list[str]
    default_sections: list[TemplateSection]
    custom_field_mappings: dict[str, str] | None = None
